tests/test_qutrit_device.py

```diff
@@ -0,0 +1,800 @@
+
+        tapes = [self.tape1] * n_tapes
+        dev.batch_execute(tapes)
+
+        assert spy.call_count == n_tapes
+
+    @pytest.mark.parametrize("r_dtype", [np.float32, np.float64])
+    def test_result(self, mock_qutrit_device, r_dtype, tol):
+        """Tests that the result has the correct shape and entry types."""
+
+        dev = mock_qutrit_device(wires=2)
+        dev.R_DTYPE = r_dtype
+
+        tapes = [self.tape1, self.tape2]
+        res = dev.batch_execute(tapes)
+
+        assert len(res) == 2
+        assert np.allclose(res[0], dev.execute(self.tape1), rtol=tol, atol=0)
+        assert np.allclose(res[1], dev.execute(self.tape2), rtol=tol, atol=0)
+        assert res[0].dtype == r_dtype
+        assert res[1].dtype == r_dtype
+
+    def test_result_empty_tape(self, mock_qutrit_device, tol):
+        """Tests that the result has the correct shape and entry types for empty tapes."""
+
+        dev = mock_qutrit_device(wires=2)
+
+        empty_tape = qml.tape.QuantumTape()
+        tapes = [empty_tape] * 3
+        res = dev.batch_execute(tapes)
+
+        assert len(res) == 3
+        assert np.allclose(res[0], dev.execute(empty_tape), rtol=tol, atol=0)
+
+
+class TestShotList:
+    """Tests for passing shots as a list"""
+
+    # TODO: Add tests for expval and sample with shot lists after observables are added
+
+    def test_invalid_shot_list(self):
+        """Test exception raised if the shot list is the wrong type"""
+        with pytest.raises(qml.DeviceError, match="Shots must be"):
+            qml.device("default.qubit", wires=2, shots=0.5)
+
+        with pytest.raises(ValueError, match="Unknown shot sequence"):
+            qml.device("default.qubit", wires=2, shots=["a", "b", "c"])
+
+    shot_data = [
+        [[1, 2, 3, 10], [(1, 1), (2, 1), (3, 1), (10, 1)], (4, 9), 16],
+        [
+            [1, 2, 2, 2, 10, 1, 1, 5, 1, 1, 1],
+            [(1, 1), (2, 3), (10, 1), (1, 2), (5, 1), (1, 3)],
+            (11, 9),
+            27,
+        ],
+        [[10, 10, 10], [(10, 3)], (3, 9), 30],
+        [[(10, 3)], [(10, 3)], (3, 9), 30],
+    ]
+
+    @pytest.mark.autograd
+    @pytest.mark.parametrize("shot_list,shot_vector,expected_shape,total_shots", shot_data)
+    def test_probs(self, shot_list, shot_vector, expected_shape, total_shots):
+        """Test a probability return"""
+        dev = qml.device("default.qutrit", wires=2, shots=shot_list)
+
+        @qml.qnode(dev)
+        def circuit(U):
+            qml.QutritUnitary(np.eye(3), wires=0)
+            qml.QutritUnitary(np.eye(3), wires=0)
+            qml.QutritUnitary(U, wires=[0, 1])
+            return qml.probs(wires=[0, 1])
+
+        res = circuit(pnp.eye(9))
+
+        assert res.shape == expected_shape
+        assert circuit.device._shot_vector == shot_vector
+        assert circuit.device.shots == total_shots
+
+        # test gradient works
+        # TODO: Add after differentiability of qutrit circuits is implemented
```

Comment:
I wasn't able to figure out exactly why `qml.jacobian` wasn't working. I will write a test that uses scalar parameters, but I don't see why a numpy array isn't working, as the `qml.jacobian` documentation says that they should work.

Reply:
What kind of error message were you getting? My initial thought is maybe it was the wrong NumPy, was it the one from PennyLane that was being imported when you tested?

Reply:
Update: I've tried using a scalar input parameter to create a unitary matrix for differentiation, but the tests are still failing. My best guess is that the reason why is because `qml.QutritUnitary` doesn't currently support decomposition, so it can't be decomposed into parametrized operations that can be differentiated.

Reply:
> What kind of error message were you getting? My initial thought is maybe it was the wrong NumPy, was it the one from PennyLane that was being imported when you tested?

Yes, I was using the PL version of numpy. I've included a screenshot of the errors below:
![image](https://user-images.githubusercontent.com/18223836/173707475-125d9cc3-e08b-4182-9341-95a2c2e26173.png)